### PR TITLE
 JAMES-2765 Avoid scrolling for quota search 

### PR DESCRIPTION
**Pull request by chibenwa**
We can easily rely on from+limit at the request level and avoid the scroll entirely.

Scrolling is expensive so should rather be avoided - scroll context is kept for a long time, waisting ES server resources.

### CODE REVIEW

File: mailbox/plugin/quota-search-elasticsearch-v6/src/main/java/org/apache/james/quota/search/elasticsearch/ElasticSearchQuotaSearcher.java [hunk 1]
```diff
@@ -57,34 +54,31 @@ public ElasticSearchQuotaSearcher(RestHighLevelClient client, ReadAliasName read
...
-            .skip(query.getOffset().getValue());
-
-        return query.getLimit().getValue()
-            .map(results::limit)
-            .orElse(results)
-            .collect(Guavate.toImmutableList());
+        try {
+            return Arrays.stream(client.search(prepareSearch(query), RequestOptions.DEFAULT)
+                .getHits()
+                .getHits())
```
Comment (blackheaven) [inline mailbox/plugin/quota-search-elasticsearch-v6/src/main/java/org/apache/james/quota/search/elasticsearch/ElasticSearchQuotaSearcher.java]: It looks weird, can you find a better name?

Comment (chibenwa) [reply mailbox/plugin/quota-search-elasticsearch-v6/src/main/java/org/apache/james/quota/search/elasticsearch/ElasticSearchQuotaSearcher.java]: For what?

Comment (trantienduchn) [reply mailbox/plugin/quota-search-elasticsearch-v6/src/main/java/org/apache/james/quota/search/elasticsearch/ElasticSearchQuotaSearcher.java]: FYI: `.getHits()` returns a `Hits` instance. `Hits.getHits()` returns an array `Hit []`

Comment (Arsnael) [reply mailbox/plugin/quota-search-elasticsearch-v6/src/main/java/org/apache/james/quota/search/elasticsearch/ElasticSearchQuotaSearcher.java]: That's ES API... not ours :trollface:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**